src/Mapbender/RoutingBundle/Component/RoutingDriver/RoutingDriver.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+namespace Mapbender\RoutingBundle\Component\RoutingDriver;
+
+use Mapbender\CoreBundle\Utils\ArrayUtil;
+
+abstract class RoutingDriver {
+
+    const INSTR_CONTINUE = 'continue';
+    const INSTR_LEFT1 = 'left1';
+    const INSTR_LEFT2 = 'left2';
+    const INSTR_LEFT3 = 'left3';
+    const INSTR_LEFT4 = 'left4';
+    const INSTR_RIGHT1 = 'right1';
+    const INSTR_RIGHT2 = 'right2';
+    const INSTR_RIGHT3 = 'right3';
+    const INSTR_RIGHT4 = 'right4';
+    const INSTR_KEEP_LEFT = 'keep-left';
+    const INSTR_KEEP_RIGHT = 'keep-right';
+    const INSTR_UTURN_LEFT = 'u-turn-left';
+    const INSTR_UTURN_RIGHT = 'u-turn-right';
+    const INSTR_FINISH = 'finish';
+    const INSTR_VIA = 'via';
+    const INSTR_ROUNDABOUT = 'roundabout';
+
+    protected string $timeField = 'time';
+    protected string $timeScale = 'ms';
+
+    abstract public function getRoute($requestParams, $configuration) : array;
+
+    abstract public function processResponse($response, $configuration);
+
+    protected function createFeatureCollection($coordinates, $type, $srid): array
+    {
+        $feature = $this->createFeature($coordinates, $type);
+        return [
+            'type' => 'FeatureCollection',
+            'crs' => [
+                'type' => 'name',
+                'properties' => [
+                    'name' => 'urn:ogc:def:crs:EPSG::' . $srid,
+                ],
+            ],
+            'features' => [$feature],
+        ];
+    }
+
+    protected function createFeature($coordinates, $type): array
+    {
+        return [
+            'type' => 'Feature',
+            'geometry' => [
+                'type' => $type,
+                'coordinates' => $coordinates,
+            ],
+            # implement properties if needed:
+            /*
+            'properties' => [
+                'key' => 'value',
+            ],
+            */
+        ];
+    }
+
+    protected function getRouteInfo($start, $destination, $distance, $time, $infoText): string
+    {
+        $search = ['{start}', '{destination}', '{length}', '{time}'];
+        $replace = [$start, $destination, $this->formatDistance($distance), $this->formatTime($time)];
+        return str_replace($search, $replace, $infoText);
+    }
+
+    protected function getRoutingInstructions($steps): array
+    {
+        return $this->addInstructionSymbols($this->translateInstructions($steps));
+    }
+
+    protected function formatTime($time): string
+    {
+        $time = gmdate('H:i', $time);
+        if (substr($time, 0, 2) == '00') {
+            $minutes = substr($time, 3, strlen($time));
+            return (substr($minutes, 0, 1) == '0') ? substr($minutes, 1, 1) . ' Min' : $minutes . ' Min';
+        }
+        return $time . ' Std.';
+    }
+
+    protected function formatDistance($distance): string
+    {
+        if (floatval($distance) >= 1000) {
+            return round((floatval($distance) / 1000.0), 2) . ' KM';
```

Comment:
KM is in uppercase and has a leading space, m is lowercase without a space, please unify